projects/cashmere/src/lib/pagination/pagination.component.ts

```diff
@@ -176,23 +204,62 @@ export class PaginationComponent implements OnChanges {
         this._goToPage((this._pageNumber || 1) + 1);
     }
 
-    private sanitize(pageNumber: any): number | null {
-        /*
-         * Validate current page, making sure it is in the valid range.
-         * values to large are set to the last page, while all other
-         * invalid values are set to 1.  If there are no pages there is
-         * also no current page.
-         */
-        if (!!this.totalPages) {
-            if (typeof pageNumber !== 'number' || isNaN(pageNumber) || !pageNumber || pageNumber < 1) {
-                pageNumber = 1;
-            }
-            if (pageNumber > this.totalPages) {
-                pageNumber = this.totalPages;
-            }
-        } else {
-            pageNumber = null;
+    /**
+     * Changes the page size so that the first item displayed on the page will still be
+     * displayed using the new page size.
+     *
+     * For example, if the page size is 10 and on the second page (items indexed 11-20) then
+     * switching so that the page size is 5 will set the third page as the current page so
+     * that the 11th item will still be displayed.
+     */
+    _changePageSize(pageSize: number) {
+        // Current page needs to be updated to reflect the new page size. Navigate to the page
+        // containing the previous page's first item.
+        const startIndex = (this.pageNumber - 1) * this.pageSize + 1;
+        this.pageSize = pageSize;
+        this.pageNumber = Math.ceil(startIndex / pageSize) || 1;
+    }
+
+    /**
+     * Updates the list of page size options to display to the user. Includes making sure that
+     * the page size is an option and that the list is sorted.
+     */
+    private _updateDisplayedPageSizeOptions() {
+        // If no page size is provided, use the first page size option or the default page size.
+        if (!this.pageSize) {
+            this._pageSize = this.pageSizeOptions.length !== 0 ? this.pageSizeOptions[0] : DEFAULT_PAGE_SIZE;
+        }
+
+        this._displayedPageSizeOptions = this.pageSizeOptions.slice();
+        if (this._displayedPageSizeOptions.indexOf(this.pageSize) === -1) {
+            this._displayedPageSizeOptions.push(this.pageSize);
+        }
+
+        // Sort the numbers using a number-specific sort function.
+        this._displayedPageSizeOptions.sort((a, b) => a - b);
+    }
+
+    private _coerceNumberValue(value: any): number {
```

Comment:
There is a CDK method that we could use for this called coerceNumberProperty(). I'll see about switching to that